File: 2. Strings/3. Valid Palindrome And Almost Palindrome/valid-palindrome.py
def valid_palindrome(s: str) -> bool:
    i, j = 0, len(s) - 1
    while i <= j:
        if s[i] == " " or not s[i].isalnum():
            i += 1
            continue
        elif s[j] == " " or not s[j].isalnum():
            j -= 1
            continue
        elif s[i].lower() != s[j].lower():
            return False
        i += 1
        j -= 1
    return True

File: 2. Strings/3. Valid Palindrome And Almost Palindrome/valid-palindrome.cpp
#include <bits/stdc++.h>
#include <iostream>

using namespace std;

class Solution {
public:
  bool isPalindrome(string s) {
    int i = 0;
    int j = s.length() - 1;

    while (i <= j) {
      if (!isalnum(s[i]) || s[i] == ' ') {
        i++;
      } else if (!isalnum(s[j]) || s[j] == ' ') {
        j--;
      } else if (tolower(s[i]) != tolower(s[j])) {
        return false;
      } else {
        i++;
        j--;
      }
    }

    return true;
  }
};

int main() { return 0; }